### PR TITLE
Dev mv code from modules to functional

### PR DESCRIPTION
**Pull request by lihuizhao**
### 移动nn.modules下的代码到nn.functional下

将interpolate、affine_grid、grid_sample、linear、layer_norm、embedding、sparse_softmax_cross_entropy、upsample、relu6
函数从nn.modules下移动到nn.functional，将对应的modules类中的forward()函数修改为调用nn.functional下的函数的形式。

### CODE REVIEW

File: python/oneflow/test/misc/test_autograd_functional.py [hunk 1]
```diff
@@ -21,6 +21,26 @@
...
+
+
+def _func_multi_scalar(x, y):
+    return (x.exp() + y.pow(2)).sum()
+
+
+def _func_scalar2tensor(x):
+    return (x, x ** 2, x ** 3)
+
+
```
Comment (marigoold) [inline python/oneflow/test/misc/test_autograd_functional.py:43]: 这些删掉

File: python/oneflow/nn/functional/sparse.py
Comment (marigoold) [inline python/oneflow/nn/functional/sparse.py:1]: 这个文件名为啥叫 sparse？

File: python/oneflow/nn/functional/upsampling.py
Comment (marigoold) [inline python/oneflow/nn/functional/upsampling.py:1]: 可以跟 interpolate 合并到一起

File: python/oneflow/nn/functional/sparse.py
Comment (lihuizhao) [reply python/oneflow/nn/functional/sparse.py:1]: 原来这个函数是放在nn/modules/sparse.py中的，它就命名为sparse，我把它迁移到nn/functional下，没有改原文件名

File: python/oneflow/nn/functional/normalization.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        normalized = flow.reshape(normalized, shape=[origin_shape[0], num_channels, -1])
+        if weight is not None:
+            normalized = normalized * weight.reshape(1, num_channels, 1)
+        if bias is not None:
+            normalized = normalized + bias.reshape(1, num_channels, 1)
+        res = flow.reshape(normalized, shape=tuple(input.shape))
+        return res
+
+
+def layer_norm(input, normalized_shape: tuple, weight=None, bias=None, eps=1e-05):
```
Comment (marigoold) [inline python/oneflow/nn/functional/normalization.py:57]: 这里可以加一下文档，参照上面的 group norm

Comment (marigoold) [reply python/oneflow/nn/functional/normalization.py:57]: 以及 type hinting

File: python/oneflow/nn/functional/linear.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import oneflow as flow
+
+
+def linear(input, weight, bias=None):
```
Comment (marigoold) [inline python/oneflow/nn/functional/linear.py:19]: type hinting

**Review by levi131 — COMMENTED** [review]
type hinting 部分没有问题
